HIP/hip-690.md

```diff
@@ -33,22 +34,113 @@ Currently, each node has an address book configuration file which ties IP addres
 
-Node addresses are configured in a JSON-formatted network file defined by the NodeAddressBook protobuf. The proposal is to add a new optional boolean property `communityNode` as a new property of the `nodeAddress` object in the address book file. If true, the flag indicates that this node belongs to a community member. If false, the flag indicates that this node belongs to a council member. If omitted the property defaults to false, ensuring backwards compatibility.
-
-The [nodeAddress protobuf definition](https://github.com/hashgraph/hedera-protobufs/blob/main/services/basic_types.proto#L1330) requires a new property `communityNode` to transmit this information.
-
-The mirrornode REST API will also expose this datum. The /network/nodes endpoint currently returns an array of `nodes` objects. A new boolean `communityNode` will be added to the `nodes` response object.
+The proposal is to add a new enumerated value `NodeOwnershipType` as a new property of the `nodeAddress` object in the address book file. 
+Possible values are `COUNCIL`=0 and `COMMUNITY`=1. 
+`COUNCIL` is the default value and indicates that the node belongs to a council member.
+`COMMUNITY` indicates that the node belongs to a community member.
+If omitted the property defaults to 0 (`COUNCIL`), ensuring backwards compatibility.
+
+The [nodeAddress protobuf definition](https://github.com/hashgraph/hedera-protobufs/blob/main/services/basic_types.proto#L1330) 
+requires a new property `nodeOwnershipType` to transmit this information.
+
+```
+/**
+* The data about a node, including its service endpoints and the Hedera account to be paid for
+* services provided by the node (that is, queries answered and transactions submitted.)
+*
+* If the <tt>serviceEndpoint</tt> list is not set, or empty, then the endpoint given by the
+* (deprecated) <tt>ipAddress</tt> and <tt>portno</tt> fields should be used.
+*
+* All fields are populated in the 0.0.102 address book file while only fields that start with # are
+* populated in the 0.0.101 address book file.
+*/
+  message NodeAddress {
+  /**
+  * The IP address of the Node with separator & octets encoded in UTF-8.  Usage is deprecated,
+  * ServiceEndpoint is preferred to retrieve a node's list of IP addresses and ports
+  */
+    bytes ipAddress = 1 [deprecated=true];
+
+  /**
+  * The port number of the grpc server for the node.  Usage is deprecated, ServiceEndpoint is
+  * preferred to retrieve a node's list of IP addresses and ports
+  */
+    int32 portno = 2 [deprecated=true];
+
+  /**
+  * Usage is deprecated, nodeAccountId is preferred to retrieve a node's account ID
+  */
+    bytes memo = 3 [deprecated=true];
+
+  /**
+  * The node's X509 RSA public key used to sign stream files (e.g., record stream
+  * files). Precisely, this field is a string of hexadecimal characters which,
+  * translated to binary, are the public key's DER encoding.  
+  */
+    string RSA_PubKey = 4;
+
+  /**
+  * # A non-sequential identifier for the node
```

Comment:
I don't think we need to copy the entire existing `NodeAddressBook` here. You can just put use ellipses to indicate existing fields like `message NodeAddressBook { ... your new field }`. See [HIP-583](https://hips.hedera.com/hip/hip-583) for an example.

Reply:
Done.